Geometry/CaloGeometry/src/CaloGeometry.cc

```diff
@@ -67,26 +66,32 @@ CaloGeometry::getSubdetectorGeometry( DetId::Detector det    ,
 static const GlobalPoint notFound(0,0,0);
 
 const GlobalPoint& 
-CaloGeometry::getPosition( const DetId& id ) const 
-{
-   const CaloSubdetectorGeometry* geom=getSubdetectorGeometry( id ) ;
-   const CaloCellGeometry* cell ( ( nullptr == geom ? nullptr : geom->getGeometry( id ) ) ) ;
-   return ( nullptr == cell ?  notFound : cell->getPosition() ) ;
+CaloGeometry::getPosition( const DetId& id ) const {
+  const CaloSubdetectorGeometry* geom = getSubdetectorGeometry( id ) ;
+  if (geom) {
+    const auto cell = (geom->getGeometry(id));
+    return (cell->getPosition());
```

Comment:
NOTE: if `cell` is now a `std::shared_ptr` then the object pointed to by the `std::shared_ptr` could be delete at the return of this function which would mean the `const GlobalPoint&` would then point to invalid memory. If this is the intend behavior, then the return type must be changed to `GlobalPoint`.